### PR TITLE
Fix array overwrite in InitNormal

### PR DESCRIPTION
**Pull request by mcflugen**
This pull request tries to fix issue #6. The code in question is immediately after the comment,

``` c
// LMV Assign fast and slow weathering portions
```

This isn't ready to be merged quite yet as I'm not 100% sure what this section of code is intended to do.

### CODE REVIEW

File: cem/cem.c [hunk 1]
```diff
@@ -4057,24 +4048,27 @@ void InitNormal(void)
...
-        }
-        else { /*if odd */
-          TypeOfRock[x][y] = 'f';
-          topography[x][y] = kCliffHeightFast;
+  {
+    const int chunk_length = Y_MAX / (1 + NUMBER_CHUNK);
+
+    for (x = 0; x < X_MAX; x++) {
+      for (n = 0; n <= 2 * NUMBER_CHUNK; n++) {
+        for (y = n * chunk_length; y < ((n + 2) * chunk_length); y++) {
```
Comment (haileyajohnson) [inline cem/cem.c]: I think rather than changing n++ to n+=2, (n+2)  should be (n+1). Otherwise chunk_length really describes half the length of a chunk. But I could be wrong, I'm not too familiar with this yet.

Comment (mcflugen) [reply cem/cem.c]: You are absolutely correct! I think I have it fixed but I just realized I forgot about CEM's period boundary conditions so I should probably fix that before merging.

**Comment by mcflugen** [comment]
I think this is ready to go now.

Added:
- `rocks.c`: contains `set_rock_blocks`, which fixes the original issue. Currently this is the only function defined here but, I think, other rock-related code should probably be moved here - just to help organize things.
- `utils.c`: contains general utility functions used by CEM - allocation of 2D arrays, applying periodic boundary conditions, etc.
- Unit tests for the new functions.

@haileyajohnson: if you want to have a look, that would be great. If not, no biggie.

File: cem/rocks.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    double *stripe = (double*)malloc(sizeof(double) * (n_fast + n_slow));
+    for (col=0; col < n_fast; col++) stripe[col] = kCliffHeightFast;
+    for (col=0; col < n_slow; col++) stripe[col + n_fast] = kCliffHeightSlow;
+    stripe_cem_matrix((void**)topography, n_rows, n_cols, sizeof(double),
+        stripe, n_fast + n_slow);
+    free(stripe);
+  }
+
+  return;
+}
```
Comment (haileyajohnson) [inline cem/rocks.c:38]: This is great, I'd been planning on pulling this out too!

**Review by haileyajohnson — APPROVED** [review]
